test/rest-adapter.test.js

```diff
@@ -245,6 +254,13 @@ describe('RestAdapter', function() {
     });
 
     describe('getFullPath', function() {
+      before(function() {
+        process.on('deprecation', NOOP);
+      });
+
+      after(function() {
+        process.removeListener('deprecation', NOOP);
+      });
```

Comment:
Nitpick: add an empty line to visually delimit after/it blocks.

Idea to consider: extract these two before/after blocks into a shared helper method.

```js
describe('getFullPath', function() {
  ignoreDeprecationsInThisBlock();

  it('returns class path + method path', /*...*/);
});
```

Thoughts?